src/libtls/tls_peer.c

```diff
@@ -1397,12 +1397,27 @@ static status_t send_certificate(private_tls_peer_t *this,
 	auth_rule_t rule;
 	bio_writer_t *certs;
 	chunk_t data;
+	tls_version_t version_min, version_max;
 
-	this->private = find_private_key(this);
+	version_min = this->tls->get_version_min(this->tls);
+	version_max = this->tls->get_version_max(this->tls);
+	if (version_max > TLS_1_1) {
+		array_t *key_types;
+
+		enumerator = tls_get_supported_key_types(version_min, version_max);
+		key_types = tls_create_common_key_types(enumerator, this->hashsig);
+		enumerator->destroy(enumerator);
+		this->private = tls_find_private_key(this->peer, this->peer_auth,
+											 key_types, NULL);
+		array_destroy(key_types);
+	}
+	else
+	{
+		this->private = find_private_key(this);
+	}
 	if (!this->private)
 	{
-		DBG1(DBG_TLS, "no TLS peer certificate found for '%Y', "
-			 "skipping client authentication", this->peer);
+		DBG1(DBG_TLS, "no usable TLS peer key found for '%Y'", this->peer);
```

Comment:
While we are searching for a key, using "certificate" in the log message might be more user accessible (but maybe use "client certificate" instead of "peer certificate").